File: frontend/lib/types/material.ts
/** Зеркало backend/app/schemas/material.py */

export interface MaterialCreate {
  name: string;
  sku?: string | null;
  category?: string | null;
  unit: string;
  min_stock?: string;
  is_active?: boolean;
}

export interface MaterialUpdate {
  name?: string | null;
  sku?: string | null;
  category?: string | null;
  unit?: string | null;
  min_stock?: string | null;
  is_active?: boolean | null;
}

export interface MaterialRead {
  id: string;
  name: string;
  sku: string | null;
  category: string | null;
  unit: string;
  min_stock: string;
  is_active: boolean;
  created_at: string;
}

File: backend/app/schemas/material.py
"""Схемы сырья и расходников."""
from __future__ import annotations

import uuid
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict, Field


class MaterialCreate(BaseModel):
    name: str = Field(min_length=1, max_length=255)
    sku: str | None = Field(default=None, max_length=100)
    category: str | None = Field(default=None, max_length=100)
    unit: str = Field(min_length=1, max_length=50)
    min_stock: Decimal = Field(default=Decimal("0"), ge=0)
    is_active: bool = True


class MaterialUpdate(BaseModel):
    name: str | None = Field(default=None, min_length=1, max_length=255)
    sku: str | None = Field(default=None, max_length=100)
    category: str | None = Field(default=None, max_length=100)
    unit: str | None = Field(default=None, min_length=1, max_length=50)
    min_stock: Decimal | None = Field(default=None, ge=0)
    is_active: bool | None = None


class MaterialRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    name: str
    sku: str | None
    category: str | None
    unit: str
    min_stock: Decimal
    is_active: bool
    created_at: datetime
